Sources/PackageModelSyntax/SyntaxEditUtils.swift

```diff
@@ -0,0 +1,225 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2014-2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Basics
+import PackageModel
+import SwiftSyntax
+import SwiftParser
+
+/// Default indent when we have to introduce indentation but have no context
+/// to get it right.
+let defaultIndent = TriviaPiece.spaces(4)
+
+extension Trivia {
+    /// Determine whether this trivia has newlines or not.
+    var hasNewlines: Bool {
+        contains { piece in
+            if case .newlines = piece {
+                return true
+            } else {
+                return false
+            }
+        }
```

Comment:
```suggestion
        contains(where: \.isNewline)
```

This should also cover `\r\n`